src/components/SettingDialog.vue

```diff
@@ -880,24 +903,25 @@ export default defineComponent({
     };
 
     const restartAllEngineProcess = () => {
-      store.dispatch("RESTART_ENGINES", { engineIds: store.state.engineIds });
+      store.dispatch("RESTART_ENGINES", { engineIds: engineIds.value });
     };
 
     const savingSetting = computed(() => store.state.savingSetting);
 
-    const samplingRateOptions: SavingSetting["outputSamplingRate"][] = [
-      "engineDefault",
-      24000,
-      44100,
-      48000,
-      88200,
-      96000,
-    ];
-    const renderSamplingRateLabel = (
-      value: SavingSetting["outputSamplingRate"]
-    ) => {
+    const samplingRateOptions = computed<SamplingRateOption[]>(() => {
+      let options = ["engineDefault", 24000, 44100, 48000, 88200, 96000];
+      if (selectedEngineId.value !== "global") {
+        options = ["inherit", ...options];
+      }
+      return options as SamplingRateOption[];
+    });
```

Comment:
stringが可変になってるのがasを使わないといけなくなってる原因かなと！
```suggestion
    const samplingRateOptions = computed<SamplingRateOption[]>(() => {
      let options: SamplingRateOption[] = [
        "engineDefault" as const,
        24000,
        44100,
        48000,
        88200,
        96000,
      ];
      if (selectedEngineId.value !== "global") {
        options = ["inherit" as const, ...options];
      }
      return options;
    });
```